OpenRA.Mods.Common/Lint/CheckVoiceReferences.cs

```diff
@@ -0,0 +1,58 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using System.Reflection;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Lint
+{
+	public class CheckVoiceReferences : ILintPass
+	{
+		public void Run(Action<string> emitError, Action<string> emitWarning, Map map)
+		{
+			foreach (var actorInfo in map.Rules.Actors)
+			{
+				var voiceSet = string.Empty;
+
+				foreach (var traitInfo in actorInfo.Value.Traits.WithInterface<ITraitInfo>())
+				{
+					var fields = traitInfo.GetType().GetFields();
+					foreach (var field in fields)
+					{
+						if (field.HasAttribute<VoiceSetReferenceAttribute>())
+							voiceSet = LintExts.GetFieldValues(traitInfo, field, emitError).FirstOrDefault();
```

Comment:
Good point @penev92.  We will need to do the checks below for each voice set on the actor, then.